google-cloud-logging/src/test/java/com/google/cloud/logging/LoggingTest.java

```diff
@@ -100,8 +104,33 @@ public void testWriteOption() {
     writeOption = WriteOption.logName(LOG_NAME);
     assertEquals(LOG_NAME, writeOption.getValue());
     assertEquals(WriteOption.OptionType.LOG_NAME, writeOption.getOptionType());
+
     writeOption = WriteOption.resource(RESOURCE);
     assertEquals(RESOURCE, writeOption.getValue());
     assertEquals(WriteOption.OptionType.RESOURCE, writeOption.getOptionType());
+
+    writeOption = WriteOption.destination(LogDestinationName.project(PROJECT_NAME));
+    LogDestinationName resource = (LogDestinationName) writeOption.getValue();
+    assertEquals(WriteOption.OptionType.LOG_DESTINATION, writeOption.getOptionType());
+    assertEquals(LogDestinationName.DestinationType.PROJECT, resource.getOptionType());
+    assertEquals(PROJECT_NAME, resource.getValue());
+
+    writeOption = WriteOption.destination(LogDestinationName.billingAccount(BILLING_NAME));
+    resource = (LogDestinationName) writeOption.getValue();
+    assertEquals(WriteOption.OptionType.LOG_DESTINATION, writeOption.getOptionType());
+    assertEquals(LogDestinationName.DestinationType.BILLINGACCOUNT, resource.getOptionType());
+    assertEquals(BILLING_NAME, resource.getValue());
+
+    writeOption = WriteOption.destination(LogDestinationName.folder(FOLDER_NAME));
+    resource = (LogDestinationName) writeOption.getValue();
+    assertEquals(WriteOption.OptionType.LOG_DESTINATION, writeOption.getOptionType());
+    assertEquals(LogDestinationName.DestinationType.FOLDER, resource.getOptionType());
+    assertEquals(FOLDER_NAME, resource.getValue());
+
+    writeOption = WriteOption.destination(LogDestinationName.organization(ORGANIZATION_NAME));
+    resource = (LogDestinationName) writeOption.getValue();
+    assertEquals(WriteOption.OptionType.LOG_DESTINATION, writeOption.getOptionType());
+    assertEquals(LogDestinationName.DestinationType.ORGANIZATION, resource.getOptionType());
+    assertEquals(ORGANIZATION_NAME, resource.getValue());
```

Comment:
nit: consider breaking it into multiple tests. it will be more visual and easy to modify in future.